src/dashboard/src/components/access/views.py

```diff
@@ -0,0 +1,112 @@
+        except AuthenticationError:
+            response = {
+                "success": False,
+                "message": "Unable to authenticate to ArchivesSpace server using the default user! Check administrative settings."
+            }
+            return django.http.HttpResponseServerError(json.dumps(response),
+                                                       content_type="application/json")
+        except ArchivesSpaceError:
+            response = {
+                "success": False,
+                "message": "Unable to connect to ArchivesSpace server at the default location! Check administrative settings."
+            }
+            return django.http.HttpResponseServerError(json.dumps(response),
+                                                       content_type="application/json")
+        except (MySQLdb.ProgrammingError, MySQLdb.OperationalError) as e:
+            response = {
+                "success": False,
+                "message": "Unable to connect to Archivist's Toolkit database; failed with error: {}".format(e),
+            }
+            return django.http.HttpResponseServerError(json.dumps(response),
+                                                       content_type="application/json")
+
+        return func(client, *args, **kwargs)
+    return wrapper
+
+
+@_authenticate_to_archivesspace
+def all_records(client, request, system=''):
+    page = request.GET.get('page', 1)
+    page_size = request.GET.get('page_size', 30)
+    search_pattern = request.GET.get('title', '')
+    identifier = request.GET.get('identifier', '')
+
+    return helpers.json_response(client.find_collections(search_pattern=search_pattern,
+                                                         identifier=identifier,
+                                                         page=page,
+                                                         page_size=page_size))
+
+
+@_authenticate_to_archivesspace
+def get_record(client, request, system='', record_id=''):
+    record_id = record_id.replace('-', '/')
+
+    try:
+        records = client.get_resource_component_and_children(record_id,
+                                                             recurse_max_level=3)
+        return helpers.json_response(records)
+    except ArchivesSpaceError as e:
+        return helpers.json_response({"error": True, "message": str(e)})
+
+
+@_authenticate_to_archivesspace
+def get_records_by_accession(client, request, system='', accession=''):
+    page = request.GET.get('page', 1)
+    page_size = request.GET.get('page_size', 30)
+
+    try:
+        records = client.find_collections_by_accession(accession, page=page, page_size=page_size)
+        return helpers.json_response(records)
+    except ArchivesSpaceError as e:
+        return helpers.json_response({"error": True, "message": str(e)})
+
+
+@_authenticate_to_archivesspace
+def record_children(client, request, system='', record_id=''):
+    if request.method == 'PUT':
+        pass
```

Comment:
It's just stubbed in right now; the intention is for this to allow new records to be added. Now that the edit form is in, I'm going to have a followup PR to add the real functionality.